metadata_backend/api/handlers/publish.py

```diff
@@ -0,0 +1,579 @@
+                    },
+                },
+            )
+
+    async def publish_submission(self, req: Request) -> Response:
+        """Update object submission specifically into published state.
+
+        :param req: PATCH request
+        :returns: JSON response containing submission ID for updated submission
+        """
+        submission_id = req.match_info["submissionId"]
+        db_client = req.app["db_client"]
+        operator = SubmissionOperator(db_client)
+        obj_op = Operator(db_client)
+        xml_ops = XMLOperator(db_client)
+
+        await operator.check_submission_exists(submission_id)
+        await self._handle_check_ownership(req, "submission", submission_id)
+
+        submission = await operator.read_submission(submission_id)
+        workflow = Workflow(submission["workflow"])
+
+        # Validate that submission has required data for publishing
+        if "datacite" in workflow.required_schemas and "doiInfo" not in submission:
+            raise web.HTTPBadRequest(reason=f"Submission '{submission_id}' must have the required DOI metadata.")
+
+        schemas_in_submission = set()
+        has_study = False
+        async for _, schema, _ in self.iter_submission_objects(submission):
+            if schema == "study":
+                has_study = True
+            if schema not in workflow.schemas:
+                reason = f"Submission of type {workflow.name} cannot have a '{schema}' but it does."
+                raise web.HTTPBadRequest(reason=reason)
+            if schema in workflow.single_instance_schemas and schema in schemas_in_submission:
+                reason = f"Submission of type {workflow.name} has more than one '{schema}', and it can have only one"
+                raise web.HTTPBadRequest(reason=reason)
+            schemas_in_submission.add(schema)
+            # TO_DO: Can only be enabled after we have unified the DAC from EGA and REMS
+            # if "dac" in workflow.required_schemas and "dac" not in submission:
+            #     raise web.HTTPBadRequest(reason=f"Submission '{accession_id}' must have DAC.")
+        if not has_study:
+            raise web.HTTPBadRequest(reason=f"Submission '{submission_id}' must have a study.")
+
+        async for _, schema, _ in self.iter_submission_objects(submission):
+            schema_dict = workflow.schemas_dict[schema]
+            if requires_or := schema_dict.get("requires_or", []):
+                for item in requires_or:
+                    if item in schemas_in_submission:
+                        break
+                else:
+                    reason = (
+                        f"'{schema}' object doesn't fulfill the submission's type {workflow.name} 'requires_or' "
+                        f"field. It should include at least one of '{requires_or}'"
+                    )
+                    raise web.HTTPBadRequest(reason=reason)
+        # TO_DO: Needs more work on the tests ^^
```

Comment:
it would be nice if TODO's are not in code, but as issues or check boxes in an issue

Reply:
Yes, I intend to work on it still, and make a new issue if any TODO's are left. I agree that it's good to have issues for them, and as long as we have issues for them, it could be beneficial to have `TODO` in the code to find it more easily.

Reply:
This needs not only tests, but also the DAC and Policy changes from #588.